plotly/plotly/plotly.py

```diff
@@ -1271,6 +1343,40 @@ def validate_credentials(credentials):
         raise exceptions.PlotlyLocalCredentialsError()
 
 
+def add_share_key_to_url(response, **payload):
+    """
+    Update plot's url to include the secret key
+
+    payload keyword agruments:
+    un (string) -- username from credentials file
+    kwargs (dict) --
+        'world_readable': (bool) make the plot public/ private
+        'sharing': ('public' | 'private' | 'secret')
+                    set privacy for the url
+
+    """
+    domain = (get_session_config().get('plotly_domain') or
+              get_config()['plotly_domain'])
+    plot_id = response['url'].replace(domain + "/~"
+                                      + payload['un'] + "/", "")
```

Comment:
:cow2: mind working some `ulrparse` magic here as well? the replace *works* but is a little fragile, i think.